### PR TITLE
[RFC] Cleanup: remove code related to pictures in the cloud-repository

### PR DESCRIPTION
**Pull request by bstoeger**
Storing pictures in the cloud never came to be, so remove the code.

Signed-off-by: Berthold Stoeger <bstoeger@mail.tuwien.ac.at>

### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This is a code-removal of picture related code like #1319. But whereas #1319 is pretty obvious (no plans to resurrect git-storage of pictures), this one not so much. I guess cloud-storage of pictures was planned as an alternative to git-storage of pictures. I don't know if there are plans of going forward with it.

### Mentions:

@atdotde @dirkhh

### CODE REVIEW

**Comment by dirkhh** [comment]
Hmm - I thought @atdotde was working on that and had plans to finish it. Did you give up on cloud storage, Robert?

File: core/imagedownloader.cpp [hunk 1]
```diff
@@ -47,7 +41,6 @@ void ImageDownloader::load(QString filename, bool fromHash)
 
 	QNetworkRequest request(url);
 	request.setAttribute(QNetworkRequest::User, filename);
-	request.setAttribute(static_cast<QNetworkRequest::Attribute>(QNetworkRequest::User + 1), fromHash);
```
Comment (atdotde) [inline core/imagedownloader.cpp:50]: This line, I am pretty sure I never wrote (nor have any idea what it does).

**Comment by bstoeger** [comment]
If there's plans to finish cloud-pictures, just close this PR.

Out of interest: how were the pictures supposed to be uploaded / synced to the cloud?

**Review by atdotde — APPROVED** [review]
I am ok with removing this code as any cloud storage for pictures never materialised. Running a cloud server with Gigs of picture data is probably not going to happen. We should rather generalise the "locate moved images" thing so it takes a URL of an html file and tries all the links on that page for pictures and hashes those. This would allow users to have all their images in a web directory and even an automatic directory listing would do the job.

File: core/imagedownloader.cpp (hunk 1 above)
Comment (bstoeger) [reply core/imagedownloader.cpp:50]: > This line, I am pretty sure I never wrote (nor have any idea what it does).

That was me. :) It transports the `fromHash` flag to the request. Indeed, not pretty at all.

**Comment by atdotde** [comment]
The idea was a simple key-value store (possibly via WebDAV): you simply dump the picture data on the server and it becomes available via the hash.

**Review by neolit123 — APPROVED** [review]

**Pull request merged**